### DISCUSSION TITLE
Custom metrics

### DISCUSSION THREAD

**Discussion by BabyGroot** [newrelic/newrelic-ruby-agent#2111] [Q&A] [answered]
Hey there,

Has anyone been able to successfully implement custom metrics in a Rails application?

I am using `newrelic_rpm-8.16.0` and incrementing a metric like this:

`NewRelic::Agent.increment_metric("Custom/Test/Test_counter", 10)` 
I get the return of `[#<NewRelic::MetricSpec 'Supportability/API/increment_metric':''>]`
However I cannot see this in my New Relic application under Metrics & Events.  All I see are other metrics for APM, RDS, Redis etc...

What am I doing wrong?

**Comment by kaylareopelle** [comment]
Hi @BabyGroot! 🌱 

Thanks for reaching out! I'm one of the maintainers of the New Relic Ruby agent. 

Where are you calling the `NewRelic::Agent.increment_metric` method? 

If it happens to be in a Rails console, the agent does not start in a Rails console by default. You'll need to set the [`autostart.denylisted_constants`](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#autostart-denylisted_constants) config in your `newrelic.yml` to an empty string. 

The executables `irb` and `rspec` also have the agent turned off by default. This can be adjusted using the [`autostart.denylisted_executables`](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#autostart-denylisted_executables) config.

**Reply by BabyGroot** [reply, marked as answer]
Hi @kaylareopelle 

Thanks for the reply.  I was calling the increment metric from the server.  Eventually I managed to find my metrics but it was a little hidden away...  I had to view all entities and click into the service and then metric explorer.  You can close the ticket.

Thanks again for your reply :)

**Reply by kaylareopelle** [reply]
I'm so glad you figured it out! Custom metrics are also known as "Timeslice Metrics" and need to be scoped by an `appName`, `appId`, or `entity.guid` to be found. You may have discovered this already from the Metrics Explorer, but in the future, if you'd like to access custom metrics from the Query Builder, you can try:

```sql
SELECT count(newrelic.timeslice.value) AS `Custom/Test/Test_counter` 
FROM Metric 
WHERE metricTimesliceName = 'Custom/Test/Test_counter' AND `appName` = 'Your App Name' 
SINCE 24 hours AGO 
TIMESERIES
```
